datafusion/core/tests/physical_optimizer/enforce_distribution.rs

```diff
@@ -2918,30 +3188,61 @@ fn parallelization_sort_preserving_merge_with_union() -> Result<()> {
     let plan_parquet = sort_preserving_merge_exec(sort_key.clone(), input_parquet);
     let plan_csv = sort_preserving_merge_exec(sort_key, input_csv);
 
+    let test_config = TestConfig::default();
+
+    // Expected Outcome:
     // should not repartition (union doesn't benefit from increased parallelism)
     // should not sort (as the data was already sorted)
+
+    // Test: with parquet
     let expected_parquet = &[
         "SortPreservingMergeExec: [c@2 ASC]",
         "  UnionExec",
         "    DataSourceExec: file_groups={1 group: [[x]]}, projection=[a, b, c, d, e], output_ordering=[c@2 ASC], file_type=parquet",
         "    DataSourceExec: file_groups={1 group: [[x]]}, projection=[a, b, c, d, e], output_ordering=[c@2 ASC], file_type=parquet",
     ];
+    test_config.run(
+        expected_parquet,
+        plan_parquet.clone(),
+        DISTRIB_DISTRIB_SORT.into(),
+    )?;
+    let expected_parquet_first_sort_enforcement = &[
```

Comment:
This looks like you added some coverage too (which is good, I just wanted to verify that was the case)